src/ucp/tag/eager.h

```diff
@@ -89,4 +91,32 @@ ucp_proto_eager_check_op_id(const ucp_proto_init_params_t *init_params,
             ucp_ep_config_key_has_tag_lane(init_params->ep_config_key));
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_add_uct_iov_elem(uct_iov_t *iov, void *buffer, size_t length,
+                     uct_mem_h memh, size_t *iov_cnt)
+{
+    iov[*iov_cnt].buffer = buffer;
+    iov[*iov_cnt].length = length;
+    iov[*iov_cnt].count  = 1;
+    iov[*iov_cnt].stride = 0;
+    iov[*iov_cnt].memh   = memh;
+    ++(*iov_cnt);
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_tag_send_am_short_iov(ucp_ep_h ep, const void *buffer, size_t length,
+                          ucp_tag_t tag)
+{
+    size_t iov_cnt      = 0ul;
+    ucp_eager_hdr_t hdr = { .super.tag = tag,
+                            .ep_id     = ucp_ep_remote_id(ep) };
+    uct_iov_t iov[2];
+
+    ucp_add_uct_iov_elem(iov, &hdr, sizeof(hdr), UCT_MEM_HANDLE_NULL, &iov_cnt);
```

Comment:
minor: no need to pass memh since it's always UCT_MEM_HANDLE_NULL now

Reply:
do you propose change `ucp_add_uct_iov_elem` args? it will change not related code

Reply:
it is not new fuction, but pls note that I moved it to other .h file to avoid cyclic dependency

Reply:
right, missed it. so ok.